lib/govuk_publishing_components/presenters/component_wrapper_helper.rb

```diff
@@ -60,7 +60,7 @@ def set_lang(lang)
       def check_id_is_valid(id)
         return if id.blank?
 
-        raise(ArgumentError, "Id (#{id}) cannot start with a number or contain whitespace and can only contain letters, digits, `_` and `-`") unless /^[a-zA-Z][\w:-]*$/.match?(id)
+        raise(ArgumentError, "Id (#{id}) cannot start with a number or contain whitespace and can only contain letters, digits, `_` and `-`") unless /\A[a-zA-Z][\w:-]*\z/.match?(id)
```

Comment:
I didn't realise this about newlines, thanks! Is it worth including a test in the component wrapper spec to check that a passed ID containing newlines is rejected?

Reply:
Good point — I agree, seems a good candidate for a regression test.